arch/x86/kvm/pkvm/pgtable.h

```diff
@@ -0,0 +1,137 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+#ifndef __PKVM_X86_PGTABLE_H
+#define __PKVM_X86_PGTABLE_H
+
+#include <linux/types.h>
+
+struct pkvm_pgtable;
+
+/**
+ * struct pkvm_pgtable_mm_ops - Page table memory management callbacks.
+ * @phys_to_virt:	Translate a virtual address to its physical address.
+ * @virt_to_phys:	Translate a physical address to its virtual address.
+ * @zalloc_page:	Allocate a page with clearing its contents.
+ * @get_page:		Increment the reference count of a given page.
+ * @put_page:		Decrement the reference count of a given page.
+ * @page_count:		Get the reference count of a given page.
+ */
+struct pkvm_pgtable_mm_ops {
+	void *(*phys_to_virt)(phys_addr_t phys);
+	phys_addr_t (*virt_to_phys)(void *vaddr);
```

Comment:
These two ops seems now can be removed. They were used for the pkvm hypervisor to access the virtual page table owned by the host. As we will use pv interfaces for ept and iommu, suppose there will be no need for the pkvm hypervisor to access such virtual page table.

Reply:
Yes, absolutely.